src/Angor/Client/Pages/Signatures.razor

```diff
@@ -380,58 +388,73 @@
         }
     }
 
-    private async Task ApproveAllSignatures()
-    {
-        var pendingSignatures = signaturesRequests.Where(s => s.TransactionHex != null && s.AmountToInvest != null && s.TimeApproved == null).ToList();
-        int totalSignatures = pendingSignatures.Count;
 
-        for (int i = 0; i < totalSignatures; i++)
+        private async Task ApproveAllSignatures()
         {
-            var signature = pendingSignatures[i];
+            isLoading = true;
+            StateHasChanged();
 
-            var operationResult = await notificationComponent.LongOperation(async () =>
+            var words = await passwordComponent.GetWalletAsync();
```

Comment:
I believe this wont work is password is empty, check this code to see how I did it to trigger the password prompt

https://github.com/block-core/angor/blob/main/src/Angor/Client/Pages/Invest.razor#L343